docs/part-7.md

```diff
@@ -0,0 +1,267 @@
+---
+id: part-7
+title: Part 7. Functional Programming Introduction
+---
+
+# Functional Programming
+
+Contributors:
+
+- Sergi Dote
+
+## Introduction
+
+Functional programming is a paradigm that puts the attention on functions. But this definition can be a little confusing since developers are coding using functions and methods and this doesn't mean they are programming following the FP paradigm.
+
+FP talks about abstract control flows, data mapping, operations, immutability, and other concepts we are reviewing in the next steps.
+
+Change our mind to a FP perspective takes a while.
+
+## What really FP is?
+
+Functional programming comes from mathematical logic, and λ-calculus. The λ-calculus helps us to describe programs as data transformations using function applications.
+
+FP teachs us to decompose a problem into simple functions and chain them to get a final result. Thinking like this, we can break the complexity of a program, and achieve any problem as a set of easy steps.
+
+In my opinion, a perfect description is what Michael Feathers said:
```

Comment:
suggestion: Rephrase by removing the expression of uncertainty to increase the impact of this sentence.
```suggestion
A perfect description is what Michael Feathers said:
```